Automattic/sensei
# Fix preview links and button for emails

Resolves #7458 

In the issue it is said, that "View Email" link is broken, but when I started testing, other places had also incorrect link.

There were changes in Gutenberg that broke our preview buttons:
- For example, changes in this [PR](https://github.com/WordPress/gutenberg/pull/56921/files) made our [EmailPreviewButton](https://github.com/Automattic/sensei/tree/trunk/assets/admin/emails/email-preview-button) not working: **Could we just delete this component and all related code?**
(I'm still looking for information about it more.)
- Here is the change for the View post link, that was originally introduced [here](https://github.com/WordPress/gutenberg/commit/9046f21a960142d3e9e5590654155d2c18ff6b53#diff-7145aecd06ef148455c1bd0c78f9b97c6f5ee5d27d82f60c0a5108299da1239b) and later moved to [this](https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/post-view-link/index.js) place with a new name. (Not sure about underlying changes.)

## Proposed Changes
* Filter the preview link.
* Remove email-preview-button component.

## Testing Instructions

1. Go to Sensei LMS -> Settings -> Emails
2. Open one of the email in the editor.
3. Make sure the preview button in the top bar works, and the URL in the sidebar is also correct and works.
4. Make a change in the email content and click "Update".
5. The notice with text "View Email" appears.
6. Click on "View Email" and make sure you see the email preview.
7. Go to Posts/Pages/Courses and makes sure their View/Preview links work properly.

## Pre-Merge Checklist
